lib/puppet/provider/user/directoryservice.rb

```diff
@@ -145,12 +145,12 @@ def self.generate_attribute_hash(input_hash)
       attribute_hash[:password] = '*'
     else
       embedded_binary_plist = get_embedded_binary_plist(attribute_hash[:shadowhashdata])
-      if embedded_binary_plist['SALTED-SHA512']
-        attribute_hash[:password] = get_salted_sha512(embedded_binary_plist)
-      else
+      if embedded_binary_plist['SALTED-SHA512-PBKDF2']
```

Comment:
Okay, that makes sense. In general I prefer things like this to be broken out into different pull requests so we can incrementally get separate changes in instead of having to okay all changes in a single PR. That being said I'm not too worried that, just an FYI here.